spring-cloud-sleuth-core/src/test/java/org/springframework/cloud/sleuth/instrument/hystrix/JavanicaITest.java

```diff
@@ -0,0 +1,125 @@
+package org.springframework.cloud.sleuth.instrument.hystrix;
+
+import static org.assertj.core.api.BDDAssertions.then;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.junit.After;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.SpringApplicationConfiguration;
+import org.springframework.cloud.netflix.hystrix.EnableHystrix;
+import org.springframework.cloud.sleuth.Span;
+import org.springframework.cloud.sleuth.TraceManager;
+import org.springframework.cloud.sleuth.instrument.DefaultTestAutoConfiguration;
+import org.springframework.cloud.sleuth.trace.TraceContextHolder;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.EnableAspectJAutoProxy;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import com.jayway.awaitility.Awaitility;
+import com.netflix.hystrix.contrib.javanica.annotation.HystrixCommand;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration(classes = {
+		JavanicaITest.JavanicaITestConfiguration.class })
+public class JavanicaITest {
+
+	@Autowired JavanicaClass javanicaClass;
+	@Autowired JavanicaDelegation javanicaDelegation;
+	@Autowired TraceManager traceManager;
+
+	@Test
+	public void should_set_span_on_an_hystrix_command_annotated_method() {
+		final Span span = givenASpanInCurrentThread();
```

Comment:
I followed the convention with the `this` - I don't like them either :)

As for what you wrote - I hate comments in the code. IMO the majority of them gets outdated the moment you write them.

Why should I write

```
// make sure this is a child span?
```

if I can write

```
make_sure_this_is_a_child_span()
```

or even better

```
then(spanCatcher.takeSpan()).isAChildSpan()
```

But I guess this is a matter of personal preference :)